loki-logback-appender/src/test/java/com/github/loki4j/logback/Loki4jAppenderTest.java

```diff
@@ -313,27 +335,38 @@ public void testRateLimitedRetry() {
                 "ts=100 l=INFO c=test.TestApp t=thread-1 | Test message 1 ")
             .build();
 
-        try { Thread.sleep(100L); } catch (InterruptedException e1) { }
+        requestBarrier.await();
         assertEquals("send", 1, sender.client.sendCount);
         assertEquals("send", expectedPayload, StringPayload.parse(sender.client.lastBatch, encoder.charset));
-        try { Thread.sleep(200L); } catch (InterruptedException e1) { }
+
+        sleep.retry.await();
+        requestBarrier.await();
         assertEquals("retry1", 2, sender.client.sendCount);
         assertEquals("retry1", expectedPayload, StringPayload.parse(sender.client.lastBatch, encoder.charset));
-        try { Thread.sleep(200L); } catch (InterruptedException e1) { }
+
+        sleep.retry.await();
```

Comment:
So, I have moved the second await to the `FailingHttpClient` but I see a lot of value in overriding sleep. The current way of testing with `Thread#sleep` and such is not reliable and extremely slow. We should not rely on `Thread#sleep` for testing, there's not guarantee that your other thread won't take longer for whatever reason in a specific run/machine, at the same time the test is just extremely slow for no reason. Right now the fastest tests are the ones where CyclicBarrier is used instead of sleep. I'd consider changing all the test in this direction rather than just making them slower for no reason. Specially with retries as they exponentially grow they'd be particularly slow and not reliable. Are you sure you want to go in that direction?